core/src/test/scala/org/apache/spark/status/AppStatusUtilsSuite.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.status
+
+import java.util.Date
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.status.api.v1.{TaskData, TaskMetrics}
+
+class AppStatusUtilsSuite extends SparkFunSuite {
+
+  test("schedulerDelay") {
+    val runningTask = new TaskData(
```

Comment:
Can we make this test case more concise and easy to read by deduplication?
For the purpose of this test case, what about the following pattern?
```
Seq(("RUNNING", 0), ("SUCCESS", 3L)).foreach { case (status, schedulerDelay) =>
 // the code from `finishedTask`
}
```

Reply:
+1

Reply:
+1

Reply:
Actually there are many different values between these 2 code blocks
```
 +        executorDeserializeTime = 5L,
 +        executorDeserializeCpuTime = 3L,
 +        executorRunTime = 90L,
 +        executorCpuTime = 10L,
 +        resultSize = 100L,
 +        jvmGcTime = 10L,
 +        resultSerializationTime = 2L,
```
I think it's OK keep the code as it is.

Reply:
Yeah, I'm inclined to keep it as they are more real.